internal/elasticsearch/index/data_stream_lifecycle_test.go

```diff
@@ -0,0 +1,239 @@
+package index_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/acctest"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccResourceDataStreamLifecycle(t *testing.T) {
+	// generate renadom name
+	dsName := sdkacctest.RandStringFromCharSet(22, sdkacctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
```

Comment:
We'll want to skip these when the backing cluster doesn't support DS lifecycles ([example](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/kibana/alerting_test.go#L34))